### PR TITLE
Creating a constructor function for FaunaClient to set default configs

### PR DESCRIPTION
**Pull request by erickpintor**

### CODE REVIEW

File: faunadb/client.go [hunk 1]
```diff
@@ -8,16 +8,45 @@ import (
...
+}
+
+func NewFaunaClient(secret string, configs ...ClientConfig) *FaunaClient {
+	client := &FaunaClient{basicAuth: basicAuth(secret)}
+
+	for _, config := range configs {
+		config(client)
+	}
+
+	if client.endpoint == "" {
```
Comment (marrony) [inline faunadb/client.go]: Just a thought: wouldn't be better to check for nullability and settings defaults on `Endpoint()` and `HTTP()` functions? That way you could make that function more clean

Comment (erickpintor) [inline faunadb/client.go]: Unless I misunderstood your point.

Comment (marrony) [inline faunadb/client.go]: ok

**Comment by marrony** [comment]
lgtm

**Comment by erickpintor** [comment]
@ed this was the breaking API change I was referring to on last sales-eng meeting.

**Pull request merged**